src/Text/XML/DSig.hs

```diff
@@ -168,7 +165,14 @@ mkSignCredsWithCert ::
 mkSignCredsWithCert mValidSince size = do
   let rsaexp = 17
   (pubkey, privkey) <- RSA.generate size rsaexp
-  validSince :: Hourglass.DateTime <- maybe (liftIO Hourglass.dateCurrent) pure mValidSince
+  let -- TODO: i think 'renderKeyInfo' is rounding time stamps to seconds (or perhaps it's
+      -- 'parseKeyInfo').  this is happening either because the standard says so (in which
+      -- case the X509 types are not restrictive enough, and shouldn't use the hourglass
+      -- DateTime type), or in x509, or in hsaml2, or in saml2-web-sso.
```

Comment:
It's actually in the standard: https://tools.ietf.org/html/rfc5280#section-4.1

> UTCTime values MUST be expressed in
   Greenwich Mean Time (Zulu) and MUST include seconds (i.e., times are
   YYMMDDHHMMSSZ), even where the number of seconds is zero.

> GeneralizedTime values MUST be
   expressed in Greenwich Mean Time (Zulu) and MUST include seconds
   (i.e., times are YYYYMMDDHHMMSSZ), even where the number of seconds
   is zero.  GeneralizedTime values MUST NOT include fractional seconds.

Reply:
4de02d0245c1b9f595e9a06a7e27b2e45c67fcd
https://github.com/vincenthz/hs-certificate/issues/119

thanks!